### ISSUE TITLE
annoying nullpo error on latest git

### ISSUE THREAD

**Issue by AnnieRuru** [HerculesWS/Hercules#2847]
**Describe the bug**
I hope I'm not the only one getting this error, every time I login my test server I keet seeing this annoying nullpo error

**To Reproduce**
- purposely reduce the hp of your character (via @heal -10000)
- logout
- login
- gets a nullpo warning

**Expected behavior**
I don't want to see any nullpo

**Screenshots**
```
[Error]: --- nullpo info --------------------------------------------
[Error]: F:\Ragnarok\Hercules\src\map\status.c:3373: 'regen->sitting' in function `status_calc_regen_pc'
[Error]: --- end nullpo info ----------------------------------------
[Error]: --- nullpo info --------------------------------------------
[Error]: F:\Ragnarok\Hercules\src\map\status.c:3598: 'regen->skill' in function `status_calc_regen_rate_pc'
[Error]: --- end nullpo info ----------------------------------------
```

**System specs (please complete the following information):**
 - OS: Windows 10 Pro 64-bit
 - Hercules Version: latest (always pull latest git)
 - Packet version: 20180620
 - Mode: Renewal
 - Client type: [2018-06-20eRagexeRE](https://rathena.org/board/topic/117168-packetver-20180620-client-release-2018-06-20eragexere/)
 - Branch: Main

**Plugins used or source modifications**
NONE - purposely unload every plugin and I still see this error !

**Additional context**

**Comment by 4144** [comment]
sadly on windows impossible to track where is issue really happened.

**Comment by AnnieRuru** [comment]
I can reproduce this error 100% of the time,
just need a `gvg_castle` mapflag

**Comment by EPuncker** [comment]
I think it is related to this commit https://github.com/HerculesWS/Hercules/commit/365ab4473c2ccf84d45cafcdb7cdc6a9a0701078#diff-08420bab8271f3965304fee42cf4b865R6056-R6061 by @hemagx 

I'm going to test it soon to confirm too

**Comment by AnnieRuru** [comment]
LMAO !!!
yes after removing those lines the error is gone

**Comment by hemagx** [comment]
@EPuncker nope, it should stay where it is we need to know the old and new map
@AnnieRuru well, the nullpo is an issue not in my code tho i need to figure out how or why this is null on player loading cause iirc it should not, also `regen.state.gc` is removed and its fine to trigger calculations from outside status.
